functions/Get-DbaAgReplica.ps1

```diff
@@ -73,7 +73,7 @@ function Get-DbaAgReplica {
         }
 
         if ($SqlInstance) {
-            $InputObject += Get-DbaAvailabilityGroup -SqlInstance $SqlInstance -SqlCredential $SqlCredential -AvailabilityGroup $AvailabilityGroup
+            $InputObject += Get-DbaAvailabilityGroup -SqlInstance $SqlInstance -SqlCredential $SqlCredential -AvailabilityGroup $AvailabilityGroup -EnableException:$EnableException.IsPresent
```

Comment:
This would not be implemented in this manner, it is not the intent of that parameter. For internal calls to public functions you add `-EnableException` and wrap the call in a try/catch. 

```suggestion
            try {
                $InputObject += Get-DbaAvailabilityGroup -SqlInstance $SqlInstance -SqlCredential $SqlCredential -AvailabilityGroup $AvailabilityGroup -EnableException
            } catch {
                    Stop-Function ...
            }
```

All of our public functions have `-EnableException` but we are not passing that on. Every public function we call in a given command should deal with "bubbling up errors" properly by using try/catch blocks.

Reply:
This is an example of why collections will cause issues when we need to deal with error handling because the call to the public function may or may not be outputting the details of the instance and AG that had issues being retrieved 🤷🏻

Reply:
@wsmelton thanks for looking at this. Do I need to create bugs for the following as well where there are similar patterns being used with the EnableException parameter (i.e. outside of try/catch blocks) - i.e. this is the pattern I saw when I raised the bug and hence the suggestion that @lancasteradam took onboard?:

Copy-DbaLogin.ps1 - lines 396, 400
Import-DbatoolsConfig.ps1 - lines 148,149
Install-DbaInstance.ps1 - lines 614
Invoke-DbaAdvancedInstall.ps1 - lines 290, 294, (line 296 is within a try/catch).
Test-DbaBuild.ps1 - lines 186, 188
Update-DbaServiceAccount.ps1 - lines 189, 235 (line 209, 213 is within a try/catch)